openedx/core/djangoapps/cors_csrf/tests/test_decorators.py

```diff
@@ -25,4 +25,6 @@ def test_ensure_csrf_cookie_cross_domain(self):
         response = wrapped_view(request)
         response_meta = json.loads(response.content.decode('utf-8'))
         assert response_meta['CROSS_DOMAIN_CSRF_COOKIE_USED'] is True
-        assert response_meta['CSRF_COOKIE_USED'] is True
+        # In Django 3.2, it's CSRF_COOKIE_USED; as of 4.0 it's CSRF_COOKIE_NEEDS_UPDATE.
+        # After upgrade to Django 4.2, delete the first clause.
+        assert response_meta['CSRF_COOKIE_USED'] is True or response_meta['CSRF_COOKIE_NEEDS_UPDATE'] is True
```

Comment:
Do we have a way of marking code for post-upgrade cleanup?

Reply:
yes we can do some thing like this 
```
import django
EXPECTED_NUM_OF_QUERIES = 5 if django.VERSION < (4, 2) else 6
```
and at the time of final `django42` PR  remove this `django.VERSION < (4, 2)`

Reply:
@irtazaakram suggested this
`
assert response_meta.get('CSRF_COOKIE_USED', response_meta.get('CSRF_COOKIE_NEEDS_UPDATE')) is True
`

Reply:
Ah, nice. I'll use `django.VERSION < (4, 0)` since that's when the change happened.